ang/prospect/services/convert-prospect-case-action.service.js

```diff
@@ -0,0 +1,79 @@
+(function (angular, $, _) {
+  var module = angular.module('prospect');
+
+  module.service('ConvertProspectCaseAction', ConvertProspectCaseAction);
+
+  function ConvertProspectCaseAction ($location, crmApi, ProspectGlobalValues) {
+    var isConvertedToProspect = false;
+
+    this.refreshData = checkIfConvertedToProspect;
+
+    /**
+     * Checks if the Action is allowed
+     */
+    this.isActionAllowed = function () {
+      return checkIfProspectingCaseTypeCategory() && !isConvertedToProspect;
+    };
+
+    /**
+     * Click event handler for the Action
+     *
+     * @param {Array} cases
+     * @param {Object} action
+     * @param {Function} callbackFn
+     */
+    this.doAction = function (cases, action, callbackFn) {
+      if (action.type !== 'contribution' && action.type !== 'pledge') {
+        return;
+      }
+
+      var caseId = cases[0].id;
+      var contactID = cases[0].client[0].contact_id;
+      var popupPath = {
+        path: 'civicrm/contact/view/' + action.type,
+        query: {
+          action: 'add',
+          reset: 1,
+          cid: contactID,
+          context: action.type,
+          caseid: caseId
+        }
+      };
+
+      return popupPath;
```

Comment:
```
return {
  // ...
};
```

Also, why is this called `doAction` if it only returns some popup configuration? All actions return configurations? If so, just call the method `getPopupParams` or similar.

Reply:
For now all services are not consistent, some do the action, and some return a url. But this will be fixed in future
https://github.com/compucorp/uk.co.compucorp.civicase/pull/228#discussion_r309537750

So keeping `doAction` for now, because thats the name across all services.